lib/Sema/ConstraintSystem.cpp

```diff
@@ -519,8 +526,22 @@ ConstraintSystem::getCalleeLocator(ConstraintLocator *locator,
                     : ConstraintLocator::Member);
   }
 
-  if (isa<UnresolvedMemberExpr>(anchor))
-    return getConstraintLocator(anchor, ConstraintLocator::UnresolvedMember);
+  if (auto *ume = dyn_cast<UnresolvedMemberExpr>(anchor)) {
+    auto *calleeLoc =
+        getConstraintLocator(ume, ConstraintLocator::UnresolvedMember);
+
+    // Handle special cases for applies of non-function types.
+    // FIXME: Consider re-designing the AST such that an unresolved member expr
+    // with arguments uses a CallExpr, which would make this logic unnecessary
+    // and clean up a bunch of other special cases. Doing so may require a bit
+    // of hacking in CSGen though.
+    if (ume->hasArguments())
```

Comment:
Nit: we usually capitalize abbreviations like these and please add more parens :)

Reply:
@xedin Presumably you mean curly braces? Should I add them to all three if statements?

Reply:
Yeah, those :) Maybe expect the deepest one, it would make it much easier to read.

Reply:
@xedin Okay cool, thanks!